arborist/engine.go

```diff
@@ -660,3 +666,114 @@ func (engine *Engine) listAuthedResources(policyIDs []string) ([]*Resource, erro
 	}
 	return resources, nil
 }
+
+// UploadModelToS3 uploads data model to S3. The function should break in the case
+// that it fails to dump the engine to JSON.
+func (engine *Engine) UploadModelToS3(cfgPath string, bucket string, key string) error {
+	bytes, err := json.Marshal(engine.toJSON())
+	if err != nil {
+		panic(err)
+	}
+
+	//Initialize AWS client and pass the config file
+	awsClient := AwsClient{}
+	if cfgPath == "" {
+		cfgPath = ConfigFile
+	}
+	err = awsClient.LoadCredentialFromConfigFile(cfgPath)
+	if err != nil {
+		fmt.Println("ERROR: Can not load aws credentials")
+		return err
+	}
+
+	// Perform upload to S3
+	return awsClient.UploadObjectToS3(bytes, bucket, key)
+}
+
+// downloadObjectFromS3 download an object from s3
+func (engine *Engine) downloadObjectFromS3(cfgPath string, bucket string, modelName string, path string) error {
+
+	// Initialize AWS client and pass the config file
+	awsClient := AwsClient{}
+	if cfgPath == "" {
+		cfgPath = ConfigFile
+	}
+
+	awsClient.LoadCredentialFromConfigFile(cfgPath)
+
+	// Perform download object from s3
+	return awsClient.DownloadObjectFromS3(bucket, modelName, path)
+}
+
+// getRootResource get the root resource
+func (engine *Engine) getRootResource(resources []ResourceJSON) *ResourceJSON {
```

Comment:
this function should be unnecessary, just use `engine.rootResource.toJSON()`

Reply:
Fixed in bc25eb4ad8ccd44d86b7a24694e3989552e789c7